### DISCUSSION TITLE
Brew install sage not working

### DISCUSSION THREAD

**Discussion by martinmeiland** [Homebrew/discussions#4853] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.16
ORIGIN: https://github.com/Homebrew/brew
HEAD: c0c8a4dd4dd19ec16deb429abd50cdf82ea33b27
Last commit: 7 days ago
Core tap JSON: 21 Oct 11:51 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.AjsQZap0L9/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.42.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.2-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  texinfo
```

### Description of issue

While trying to install sage the install seem to stop working after downloading and mounting using hdiutil. I have run the installer with the debug option, the output can be seen below. Does anyone have any idea how to fix this?

==> Downloading https://formulae.brew.sh/api/cask.jws.json
/usr/bin/env /usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --user-agent Homebrew/4.1.16\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 13.5.2\)\ curl/8.1.2 --header Accept-Language:\ en --fail --remote-time --output /Users/martinmeiland/Library/Caches/Homebrew/api/cask.jws.json --location --time-cond /Users/martinmeiland/Library/Caches/Homebrew/api/cask.jws.json --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.1.16\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 13.5.2\)\ curl/8.1.2 --header Accept-Language:\ en --fail --compressed --speed-limit 100 --speed-time 5 https://formulae.brew.sh/api/cask.jws.json
 
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

/usr/local/Homebrew/Library/Homebrew/brew.rb 

(Cask::CaskLoader::FromAPILoader): loading sage

==> Cask::Installer#install

==> Printing caveats

==> Cask::Installer#fetch

==> Downloading https://github.com/3-manifolds/Sage_macOS/releases

/download/v2.1.0/SageMath-10.1_x86_64.dmg
/usr/bin/env /usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.1.16\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 13.5.2\)\ curl/8.1.2 --header Accept-Language:\ en --retry 3 --fail --location --silent --head https://github.com/3-manifolds/Sage_macOS/releases/download/v2.1.0/SageMath-10.1_x86_64.dmg
Already downloaded: /Users/martinmeiland/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg

==> Checking quarantine support

/usr/bin/env /usr/bin/xattr -h
/usr/bin/env /usr/bin/swift -target x86_64-apple-macosx13 /usr/local/Homebrew/Library/Homebrew/cask/utils/quarantine.swift

==> Quarantine is available.

==> Verifying Gatekeeper status of /Users/martinmeiland/Library/Caches

/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/martinmeiland/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg

==> /Users/martinmeiland/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg is quarantined

==> Verifying checksum for '6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg'
/usr/bin/env hdiutil imageinfo -format /Users/martinmeiland/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg

==> Installing Cask sage
==> Cask::Installer#stage
==> Extracting primary container
==> Using container class UnpackStrategy::Dmg for /Users/martinmeiland

/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg
/usr/bin/env hdiutil attach -plist -nobrowse -readonly -mountrandom /private/tmp/d20231021-3573-1jfxrlz /Users/martinmeiland/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg
/usr/bin/env hdiutil convert -format UDTO -o /private/tmp/d20231021-3573-1jfxrlz/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.cdr /Users/martinmeiland/Library/Caches/Homebrew/downloads/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.dmg
/usr/bin/env hdiutil attach -plist -nobrowse -readonly -mountrandom /private/tmp/d20231021-3573-1jfxrlz /private/tmp/d20231021-3573-1jfxrlz/6d611958d6ef5c9bd47547c1d1888462820b8c6c5a549742d7fad505d6a47adb--SageMath-10.1_x86_64.cdr
/usr/bin/env find . -print0
/usr/bin/env mkbom -s -i /private/tmp/20231021-3573-19n99o4.list -- /private/tmp/20231021-3573-19qqap9.bom
/usr/bin/env ditto --bom /private/tmp/20231021-3573-19qqap9.bom -- /private/tmp/d20231021-3573-1jfxrlz/dmg.yHRwHT /private/tmp/d20231021-3573-w0y7v7

### Relevant casks

sage

**Comment by krehel** [comment]
Hi @martinmeiland -

How long did you wait for the install to complete?  I tried to replicate the issues, but I was unable to.  

I will note that the application maybe took 10 minutes to install and did seem like it was no longer working at one point, but it did finish eventually.

**Reply by martinmeiland** [reply]
Sorry, for not responding have been really busy for a couple of months. I waited for more than 1 hour and it did not install, I just factory reset my mac and that seems to have fixed the issue. I don't know what changed or why it works now, but many thanks for replying.
